### ISSUE TITLE
bootstrap.el should not provide features

### ISSUE THREAD

**Issue by tarsius** [tgbugs/orgstrap#1]
Test and setup files should not `provide` a feature. Providing a feature is only suitable for libraries that are loaded by other libraries/packages or by the end user using `require`. For `require` to be able to do its job, the library that provides the feature has to be located on the `load-path`, but test files should *not* be located on the `load-path`.

You are probably already loading these setup files using something `(load "relative/path/to/it")` and can just remove the `provide` form.  If not, then you probably just have to replace a `require` with a such a `load` form somewhere.

One reason why test files should not `provide` a feature and/or reside on the `load-path` is that authors tend to name those files just `tests.el` or something like that.  If only one package does that, then that works, but it only two out of several thousands and things start to get problematic. In fact I usually don't notice such issues until there actually are two packages that provide the exact same "testing feature".

<sub>I have written similar messages many many times so I now [use a template](https://help.github.com/en/github/writing-on-github/working-with-saved-replies). I try to adjust it to match each individual case but it might still not apply 100% in all cases.</sub>

**Comment by tgbugs** [comment]
Thank you for the report! I've fixed the issue, but I'm not sure that it was entirely applicable since bootstrap.el was only pulled in as part of init.el and not distributed (nor intended to be distributed) with orgstrap.el on melpa.

**Comment by tarsius** [comment]
Thanks!

Yes, when using Melpa then this is not an issue. But when not using Melpa, then it has the potential to be an issue. :grin:

For example when using my `borg` package manager or another alternative package manager. Or when installing packages manually. Or when automatically extracting metadata for the Emacsmirror (which can be looked at using my `epkg` package).

**Comment by tarsius** [comment]
I just noticed that I forgot to mention earlier that `init.el` has the same issue. Could you please remove the provide form from that too?

**Comment by tgbugs** [comment]
Yes will do. That one definitely doesn't need it, not sure why I put it there in the first place.

**Comment by tgbugs** [comment]
init.el is fixed by https://github.com/tgbugs/orgstrap/commit/bf64fda2eb881728ac5fd04932f64cd0c3b7918b. packages.el still has a provide form, will it cause issues?

**Comment by tarsius** [comment]
> packages.el still has a provide form, will it cause issues?

I would recommend that you remove that as well.

The reason I noticed the `bootstrap` and `init` features is that some other package also provided those features. (It should not do that for the same reasons your package should not do it.) There currently is no other package that provides the feature `packages`, which is why I did not notice. But that just means that your package currently flies under the radar because it is the only one that provides this feature without being supposed to do so.

It makes no more sense for your package to provide `bootstrap`, `init` and/or `packages` than for any other packages and you therefore should not do it because sooner or later someone else won't realize this potential conflict and start providing one of these features. It happens quite frequently.
